llext: misc preparatory improvements

lyakh opened the pull request.

lyakh committed:
llext: skip first dummy symbol table entry

The first entry in the symbol table is empty, skip it.

Signed-off-by: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>

lyakh committed:
llext: use provided size for more generic code

Use .sh_entsize instead of sizeof(elf_rel_t) to make code suitable
for both REL and RELA types.

Signed-off-by: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>

lyakh committed:
llext: clarify the use of loop variables, simplify code

Several cosmetic changes with no change in functionality:

The pos variable in multiple functions is used as a loop variable -
it's initialised before the loop starts and then it's incremented for
each loop iteration. Move it to the loop header for readability.

"return ret" is clearer than "goto out" where the "out" label does
nothing but "out: return ret" because one sees immediately that the
function is terminated at that location with no further actions
without the need to check the "out" label.

k_heap_free(heap, NULL) is valid, no need to check the address to be
freed for NULL.

Object counters can be simple (unsigned) integers, no need to make
them size_t or elf_word.

"return 0" is simpler than "return ret" because it shows the return
value immediately without the need to check what it can be at that
location.

Signed-off-by: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>

pillo79 reviewed (changes requested):
The `str_cnt` variable is used to count the _tables_ that were found and break early. I'd say this is the right moment to refactor it to `tab_cnt` or `sect_cnt`. Everything else LGTM!

lyakh committed:
llext: (cosmetic) rename a variable to match its role

str_cnt in llext_find_tables() is actually used to count sections,
rename it to better match that usage.

Signed-off-by: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>

lyakh commented:
> The `str_cnt` variable is used to count the _tables_ that were found and break early. I'd say this is the right moment to refactor it to `tab_cnt` or `sect_cnt`. Everything else LGTM!

@pillo79 since it isn't changed by this PR I made it an additional commit
